File: java/muse/algorithms/search/BinarySearchTest.java
import muse.algorithms.search.BinarySearch;
import muse.util.SequenceBuilder;
import muse.util.TestRunner;

public class BinarySearchTest {

    public static boolean testBinarySearch() {
        int size = 32768;

        Integer[] arr = new Integer[size];
        SequenceBuilder.packIncreasing(arr);

        if (BinarySearch.find(arr, -1) != size) {
            return false;
        }

        if (BinarySearch.find(arr, 2_147_483_647) != size) {
            return false;
        }

        for (int i = 0; i < size; i++) {
            if (BinarySearch.find(arr, arr[i]) != i) {
                return false;
            }
        }

        return true;
    }

    public static void main(String[] args) {
        TestRunner.pick(BinarySearchTest::testBinarySearch);
    }
}

File: php/muse/algorithms/search/BinarySearchTest.php
<?php
    require_once "muse/algorithms/search/BinarySearch.php";
    require_once "muse/util/SequenceBuilder.php";
    require_once "muse/util/TestRunner.php";

    use muse\algorithms\search\BinarySearch;
    use muse\util\SequenceBuilder;
    use muse\util\TestRunner;

    function testBinarySearch(): bool {
        $size = 32768;

        $arr = array_pad(array(), $size, 0);
        SequenceBuilder\packIncreasing($arr);

        if (BinarySearch\find($arr, -1) != $size) {
            return false;
        }

        if (BinarySearch\find($arr, 2147483647) != $size) {
            return false;
        }

        foreach ($arr as $i => $v) {
            if (BinarySearch\find($arr, $v) != $i) {
                return false;
            }
        }

        return true;
    }

    if (count(debug_backtrace()) == 0) {
        TestRunner\pick(testBinarySearch);
    }
?>
